discord/bot.py

```diff
@@ -196,105 +196,161 @@ def get_application_command(
-                for key, more_keys in {
-                    key:more_keys
-                    for key, more_keys in keys_to_check.items()
-                    if key in as_dict.keys()
-                    if key in match.keys()
-                }.items():
-                    if key == "options":
-                        for i, option_dict in enumerate(as_dict[key]):
-                            if command.name == "recent":
-                                print(option_dict, "|||||", match[key][i])
-                            for key2 in more_keys:
-                                pendingVal = None
-                                if key2 in option_dict.keys():
-                                    pendingVal = option_dict[key2]
-                                    if pendingVal == False or pendingVal == []: # Registered commands are not available if choices is an empty array or if autocomplete is false
-                                        pendingVal = None
-                                matchVal = None
-                                if key2 in match[key][i].keys():
-                                    matchVal = match[key][i][key2]
-                                    if matchVal == False or matchVal == []: # Registered commands are not available if choices is an empty array or if autocomplete is false
-                                        matchVal = None
-
-                                if pendingVal != matchVal:
-                                    # When a property in the options of a pending global command is changed
-                                    needs_bulk = True
-                    else:
-                        if as_dict[key] != match[key]:
-                            # When a property in a pending global command is changed
-                            needs_bulk = True
+            if as_dict and match:
+                # Check if something in the type: 1 sub_command is changed
+                is_same = compare_commands(0, as_dict, match)
             else:
-                # When a name of a pending global command is not registered in Discord
-                needs_bulk = True
+                # A new command has been added
+                is_same = False
+
+            if not is_same:
+                needs_update = True
 
             commands_to_bulk.append(as_dict)
-        
-        for name, command in registered_commands_dict.items():
+
+        for name in registered_commands_dict.keys():
             if not name in global_pending_application_commands_dict.keys():
                 # When a registered global command is not available in the pending global commands
-                needs_bulk = True
+                needs_update = True 
+
+        await self.register_commands(commands_to_bulk, needs_update)
+
+    async def register_commands(self, commands, needs_update) -> None:
```

Comment:
```suggestion
    async def register_commands(
        self,
        commands: List[ApplicationCommand],
        needs_update: bool = False,
    ) -> None:
```
Maybe also document needs_update